### PR TITLE
Validate Story Slug

### PR DESCRIPTION
**Pull request by SarvaBharan**
Added feature to check if story slug is valid or not. #68

### CODE REVIEW

File: server/src/lib/utils/validator.ts [hunk 1]
```diff
@@ -92,7 +107,7 @@ export function validateJson(
     finalErrors = validateSlug(data.slug,uniqueSlugs,validate.errors || [])
   }
   if(data.body) {
-    finalErrors = validateBody(data.body, validate.errors || [])
+    finalErrors = finalErrors.concat(validateBody(data.body, validate.errors || []));
```
Comment (ddhar28) [inline server/src/lib/utils/validator.ts:110]: why do we need the `concat`?  `validateBody` already returns the updated array right?

Comment (ddhar28) [inline server/src/lib/utils/validator.ts:110]: why do we need the `concat`?  `validateBody` already returns the updated array right?

Comment (SarvaBharan) [reply server/src/lib/utils/validator.ts:110]: `validateSlug` function's value was overridden if the `validateBody` function returned with no error. Basically `finalErrors` value holding from `validateSlug` function was overridden to `[]` @ddhar28

File: server/src/lib/utils/validator.ts [hunk 2]
```diff
@@ -65,7 +65,22 @@ function validateBody(body: string, errors: Array<ajv.ErrorObject>) {
   return errors
 }
 
+function isValidSlug(slug: string) {
+  let slugRegex = /^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$/;
```
Comment (Robert-Christopher) [inline server/src/lib/utils/validator.ts:69]: Check if some library already has this feature, refer to some typescript repos in Quintype, we might be using some library

Comment (Robert-Christopher) [reply server/src/lib/utils/validator.ts:69]: We can go with regexp for now

Comment (SarvaBharan) [reply server/src/lib/utils/validator.ts:69]: Searched, I could find only “slugify” function and checks on, if slug has got value.

**Pull request merged**